### PR TITLE
gh-120674: protect multi-line macro expansions in `Modules/_testbuffer.c` and `Modules/_testcapimodule.c`

### PR DESCRIPTION
**Pull request by picnixz**
(skip news is needed for this one)

* Issue: gh-120674

### CODE REVIEW

File: Modules/_testcapimodule.c [hunk 1]
```diff
@@ -1024,12 +1044,12 @@ test_capsule(PyObject *self, PyObject *Py_UNUSED(ignored))
         static char buffer[256];
 #undef FAIL
 #define FAIL(x) \
-        { \
+    do { \
```
Comment (picnixz) [inline Modules/_testcapimodule.c]: If needed, I can add an additional indentation level here

File: Modules/_testcapimodule.c [hunk 2]
```diff
@@ -1955,8 +1975,7 @@ py_w_stopcode(PyObject *self, PyObject *args)
 
 
 static PyObject *
-test_pythread_tss_key_state(PyObject *self, PyObject *args)
-{
+test_pythread_tss_key_state(PyObject *self, PyObject *args) {
```
Comment (picnixz) [inline Modules/_testcapimodule.c]: This change was not intended (I'll revert it).

File: Modules/_testcapimodule.c [hunk 3]
```diff
@@ -1069,7 +1089,7 @@ test_capsule(PyObject *self, PyObject *Py_UNUSED(ignored))
             PyErr_Clear();
     }
 
-  exit:
+exit:
```
Comment (picnixz) [inline Modules/_testcapimodule.c]: This change was not intended (I'll revert it).

File: Modules/_testbuffer.c [hunk 4]
```diff
@@ -24,11 +24,13 @@ static PyTypeObject NDArray_Type;
...
 
 #define CHECK_LIST_OR_TUPLE(v) \
-    if (!PyList_Check(v) && !PyTuple_Check(v)) { \
-        PyErr_SetString(PyExc_TypeError,         \
-            #v " must be a list or a tuple");    \
-        return NULL;                             \
-    }                                            \
+    do { \
+        if (!PyList_Check(v) && !PyTuple_Check(v)) { \
+            PyErr_SetString(PyExc_TypeError,                  \
```
Comment (vstinner) [inline Modules/_testbuffer.c]:
```suggestion
            PyErr_SetString(PyExc_TypeError, \
```

**Review by vstinner — APPROVED** [review]
LGTM if you fix a minor consistency issue :-)

**Comment by vstinner** [comment]
> LGTM if you fix a minor consistency issue :-)

I applied my minor suggestion and enabled automerge.

**Comment by picnixz** [comment]
Thank you for the consistency nitpick! Some parts of the code align the '\\' and others don't. In this PR, my changes actually used the following convention by deciding not to align the '\\' **except** when they are used for breaking the same expression (e.g., a function call). So thank you! 

By the way, in the future, I don't mind people comitting directly onto my branch if I don't reply directly.

**Pull request merged**